docs/models

```diff
@@ -0,0 +1,120 @@
+The PythonItemModel QML object provides the ability to implement a QML data
+model in Python. This is useful because a Python backend can then
+programatically manipulate the Python model directly from Python in a Pythonic
+fashion, and the QML frontend can be implemented with a standard view in a
+standard QML fashion.
+
+Right now the implementation only supports a limited set of read-only
+operations from the QML end (so a view in QML can only read from the model) but
+the architecture could quite easily be extended to support full read-write
+models in future.
+
+Example:
+
+	import io.thp.pyotherside 1.5
+	
+	Rectangle {
+	    ListView {
+	        model: model
+	        delegate: Row {
+	            Text { text: model.display }
+	        }
+	        anchors.fill: parent
+	    }
+	    PythonItemModel {
+	        id: model
+	    }
+	    Python {
+	        id: py
+	        Component.onCompleted: {
+	            addImportPath(Qt.resolvedUrl('.'));
+	            importModule('main', function() {
+	                py.call('main.get_model', [], function(result) {
+	                    // this should change to:
+	                    //     model.change_model(model_constructor);
+	                    // in future as a property is inappropriate
+	                    model.model = result;
+	                });
+	            });
+	        }
+	    }
+	}
+
+and then in main.py:
+
+    import pymodel
+    
+    def get_model(bridge):
```

Comment:
Continuing from the example above, there's no "bridge" parameter supplied in the QML example code above.